community/values/src/main/java/org/neo4j/values/storable/UTF8StringValue.java

```diff
@@ -158,21 +158,21 @@ public int computeHash()
     @Override
     public TextValue substring( int start, int length )
     {
-        int s = Math.min( start, length() );
-        int e = Math.min( s + length, length() );
-        byte[] values = bytes;
-        if ( s >= e )
+        if ( start < 0 || length < 0)
         {
-            return StringValue.EMTPY;
+            throw new IndexOutOfBoundsException( "Cannot handle negative start index nor negative length" );
         }
```

Comment:
We could possibly do a quick return of `StringValue.EMPTY` here if `length == 0`.